### PR TITLE
render parser errors using the diagnostics interface

### PR DESCRIPTION
**Pull request by reiner-dolp**
image
image
image

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -192,9 +191,18 @@ fn cmd_parsetest(path: &PathBuf) -> Result<(), Error> {
...
     match parser.parse() {
         Ok(_) => Ok(()),
-        // TODO: generate error message per spec
-        Err(err) => {
-            eprintln!("error: {}", err);
+        Err(parser_error) => {
+            // TODO: context.error should do this match automatically through
+            // generic arguments
+            match parser_error {
+                MaybeSpanned::WithSpan(spanned) => context.error(Spanned {
```
Comment (joshuabach) [inline src/main.rs]:
```suggestion
                MaybeSpanned::WithSpan(spanned) => spanned.map(Box::new)
```

Comment (joshuabach) [reply src/main.rs]: This also replaces the next 3 lines

File: src/parser.rs [hunk 2]
```diff
@@ -207,7 +198,7 @@ where
         self.peek()?;
         match self.lexer.next() {
             Some(token) => Ok(token),
-            None => self.eof_token.clone().ok_or(SyntaxError::MissingEOF),
+            None => self.eof_token.clone().ok_or(MaybeSpanned::WithoutSpan(SyntaxError::MissingEOF)),
```
Comment (joshuabach) [inline src/parser.rs]:
```rust
use MaybeSpanned::*;
```

File: src/main.rs (hunk 1 above)
Comment (reiner-dolp) [reply src/main.rs]: This suggestion does not work.

**Pull request merged**